osdep/terminal-unix.c

```diff
@@ -411,8 +412,9 @@ static void *terminal_thread(void *ptr)
     bool stdin_ok = read_terminal; // if false, we still wait for SIGTERM
     while (1) {
         getch2_poll();
-        struct pollfd fds[2] = {
+        struct pollfd fds[3] = {
             { .events = POLLIN, .fd = death_pipe[0] },
+            { .events = POLLIN, .fd = stop_cont_pipe[0] },
```

Comment:
why not reuse death_pipe and make it take three values?

Reply:
I considered it, but it'll lead to some subtle semantic changes. Currently any activity on the death_pipe breaks the loop, but the stop_cont pipe only reacts to POLLIN.

There's also the (mostly theoretical) possibility of the pipe filling up due to lots of signals being delivered rapidly and some of it getting discarded due to full pipe. Giving each signal it's own pipe gets around certain signals getting "lost" etc.

I can still change to using a single pipe if you prefer that, since the "problems" are mostly theoretical.

Reply:
Better leave it then.